src/Language/PureScript/Errors.hs

```diff
@@ -1306,6 +1316,25 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
         , line "Instance declarations of this type class are disallowed."
         ]
 
+    renderSimpleErrorMessage UnsupportedRoleDeclaration =
+      line $ "Role declarations are only supported for data types, not for type synonyms nor type classes."
+
+    renderSimpleErrorMessage (RoleDeclarationArityMismatch name expected actual) =
+      line $ T.intercalate " "
+        [ "The type"
+        , markCode (runProperName name)
+        , "expects"
+        , T.pack (show expected)
+        , if expected > 1 then "arguments" else "argument"
```

Comment:
When there are no arguments, the word "argument" should be pluralized, so I think the condition we want here is `if expected == 1 then "argument" else "arguments"`